src/util/format-ref.util.ts

```diff
@@ -0,0 +1,18 @@
+import { Reference } from "../model/reference.model";
+import { getDate } from "./get-date.util";
+
+/**
+ * Format a given reference as a valid markdown url
+ */
+export function formatReference (ref: Reference): string {
+    const author: string = ref.author[0].family;
+    const year: number = getDate(ref).getFullYear();
+    const title: string = ref.title.split(" ").join("+");
+
+    let url: string = "";
+    if (ref.DOI && ref.DOI !== "") url = `http://dx.doi.org/${ref.DOI}`;
+    else if (ref.URL && ref.URL !== "") url = ref.URL;
+    else url = `https://scholar.google.com/scholar?q=${title},+${author}`;
```

Comment:
Please use brackets to improve readability and avoid errors [like this](https://embeddedgurus.com/barr-code/2014/03/apples-gotofail-ssl-security-bug-was-easily-preventable/).

Reply:
ref.DOI should be escaped. Please look for "encodeURI" which I think is what you need for URLs.

Sale for title and author which should be escaped too.